code/game/objects/items/devices/taperecorder.dm

```diff
@@ -11,53 +11,111 @@
 	slot_flags = ITEM_SLOT_BELT
 	custom_materials = list(/datum/material/iron=60, /datum/material/glass=30)
 	force = 2
-	throwforce = 0
+	throwforce = 2
+	speech_span = SPAN_TAPE_RECORDER
 	drop_sound = 'sound/items/handling/taperecorder_drop.ogg'
 	pickup_sound = 'sound/items/handling/taperecorder_pickup.ogg'
-	var/recording = 0
-	var/playing = 0
+	var/recording = FALSE
+	var/playing = FALSE
 	var/playsleepseconds = 0
 	var/obj/item/tape/mytape
 	var/starting_tape_type = /obj/item/tape/random
-	var/open_panel = 0
-	var/canprint = 1
+	var/open_panel = FALSE
+	var/canprint = TRUE
+	var/list/icons_available = list()
+	var/radial_icon_file = 'icons/obj/radial_taperecorder.dmi'
+	///Whether we've warned during this recording session that the tape is almost up.
+	var/time_warned = FALSE
+	///Seconds under which to warn that the tape is almost up.
+	var/time_left_warning = 60 SECONDS
+	///Sound loop that plays when recording or playing back.
+	var/datum/looping_sound/tape_recorder_hiss/soundloop
 
 
 /obj/item/taperecorder/Initialize(mapload)
 	. = ..()
 	if(starting_tape_type)
 		mytape = new starting_tape_type(src)
+	soundloop = new(src, FALSE)
 	update_icon()
 	become_hearing_sensitive()
 
+/obj/item/taperecorder/proc/readout()
+	if(mytape)
+		if(playing)
+			return "<span class='notice'><b>PLAYING</b></span>"
+		else
+			var/time = mytape.used_capacity / 10 //deciseconds / 10 = seconds
+			var/mins = round(time / 60)
+			var/secs = time - mins * 60
+			return "<span class='notice'><b>[mins]</b>m <b>[secs]</b>s</span>"
+	return "<span class='notice'><b>NO TAPE INSERTED</b></span>"
+
+/obj/item/taperecorder/AltClick(mob/user)
+	. = ..()
+	if(!can_interact(user))
+		return
+	play()
```

Comment:
Alt click to play is a confusing interaction. I'm not a fan of alt click and typically I would say that a radial menu with either play or eject would be better, but if you are going to use alt-click, alt-click should be for eject and attack_self should be for play.

It looks like you do have a radial menu already, so this AltClick interaction probably isn't necessary at all.